core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -291,6 +309,7 @@ final class ShuffleBlockFetcherIterator(
             // This needs to be released after use.
             buf.retain()
             remainingBlocks -= blockId
+            logFetchOnCompletionIfSlow()
             blockOOMRetryCounts.remove(blockId)
             results.put(new SuccessFetchResult(BlockId(blockId), infoMap(blockId)._2,
               address, infoMap(blockId)._1, buf, remainingBlocks.isEmpty))
```

Comment:
Make `remainingBlocks.isEmpty` into a local variable `networkReqDone` and use that for both `SuccessFetchResult` and whether to invoke `logFetchOnCompletionIfSlow` (and you can remove the `remainingBlocks.isEmpty` in that method)